CMakeLists.txt

```diff
@@ -41,11 +41,12 @@ if(NOT DEFINED CMAKE_INSTALL_CMAKEDIR)
     set(CMAKE_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/nanopb")
 endif()
 
+# Determine Python module installation path
 if (NOT nanopb_PYTHON_INSTDIR_OVERRIDE)
     find_package(Python REQUIRED COMPONENTS Interpreter)
     execute_process(
         COMMAND ${Python_EXECUTABLE} -c
-            "import os.path, sys, sysconfig; print(os.path.relpath(sysconfig.get_path('purelib'), start=sys.prefix))"
+            "import os.path, sys, sysconfig; print(sysconfig.get_path('purelib'))"
```

Comment:
This is equivalent to [`Python_SITELIB`](https://cmake.org/cmake/help/latest/module/FindPython.html#result-variables).

Reply:
Good point

Reply:
For some reason it is not actually equivalent. When I install with the current `sysconfig.get_path('purelib')`, it installs files to `/usr/local/lib/python3.10/dist-packages/nanopb/`. When I use `Python_SITELIB`, it installs to `/usr/lib/python3/dist-packages/nanopb/`.

Probably it should follow the installation prefix given to CMake instead. I think that is what the current code in master branch tries to do, but it produces an invalid path for me.

Reply:
I assume your cmake python find module use `distutils.sysconfig.get_python_lib(plat_specific=False,standard_lib=False)` instead of `sysconfig.get_path('purelib')`.

Reply:
I think this is how we would want this to work:

1. When installed system-wide, install to Python `site-packages` in the CMake installation prefix. I guess `dist-packages` is ok too, though `site-packages` is probably more correct.
2. When installed to a non-system prefix, install `nanopb` module into subdirectory of `bin` so that it can be found by `nanopb_generator.py` wrapper with relative path.

I wonder what is the cleanest way to distinguish between 1. and 2. in CMake.

Reply:
Maybe the python scripts inside `bin` should use a relative path to import the package. This ensure that the bin files always import the correct version and doesn't requires symbolic links. The relative path could be determined by CMake.

The user should use `DESTDIR` to install the files into its home directory because this will be prepend to all paths.

Reply:
Relative path will work fine for installation into home directory or project specific directory. But there are users who wish to install system-wide, and then it makes sense to install to the usual Python module location.

Reply:
I would install the Python module to the Python module location always. The user installation will move the Python module location below `DESTDIR`:
```
/home/user/nanopb/lib/python3/dist-packages/nanopb/generator/proto
/home/user/nanopb/usr/local/bin/nanopb_generator.py
```

The `nanopb_generator.py` will import the Python module relative to its location:
```
p = os.path.abspath(os.path.dirname(__file__) + "/../../lib/python3/dist-packages")
import sys
sys.path.insert(0, p)
```

Reply:
Hmm, that seems reasonable as it will still search the system folders also.
However I'm not sure if the location under DESTDIR will always be exactly that.
So it will need some kind of search-replace with CMake to substitute the actual relative path.

Reply:
Yes like the manipulation of the bat files. CMake can determine the [relative path](https://cmake.org/cmake/help/latest/command/file.html#relative-path) between `PYTHON_INSTDIR` and `CMAKE_INSTALL_FULL_BINDIR`. You could use [`configure_file`](https://cmake.org/cmake/help/latest/command/configure_file.html) to replace `@VAR@` inside a file.